### PR TITLE
Sidebar: Load domain warnings asynchronously

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request is a follow-up of #15341 that aims at loading the domain warnings displayed in the sidebar asynchronously. The goal is to reduce the size of the main Javascript bundle by creating a new chunk only for this component: before it would weight 5.96MB (633.5KB zipped). However after this change it weights 5.69MB (600.6KB zipped).

screenshot

#### Testing instructions
 
1. Run `git checkout update/domain-warnings` and start your server, or open a [live branch](https://calypso.live/?branch=update/domain-warnings)
2. Open any page from the `My Sites` section, assuming you have a site selected
3. Check that domain warnings display as before in the sidebar

#### Reviews
 
- [x] Code
- [ ] Product
- [ ] Tests

### CODE REVIEW

File: client/my-sites/current-site/domain-warnings.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	return (
+		<div>
+			<QuerySiteDomains siteId={ selectedSite.ID } />
+
+			<DomainWarnings
+				isCompact
+				selectedSite={ selectedSite }
+				domains={ domains }
+				ruleWhiteList={ [
```
Comment (yurynix) [inline client/my-sites/current-site/domain-warnings.jsx]: That creates new array each `render()`

**Comment by yurynix** [comment]
Looks good and works as described, LGTM 👍 
Had only a minor comment, but we can merge without it.

File: client/my-sites/current-site/domain-warnings.jsx (hunk 1 above)
Comment (stephanethomas) [reply client/my-sites/current-site/domain-warnings.jsx]: I'll update that.

**Comment by aidvu** [comment]
Thanks for this @stephanethomas! LGTM.

**Comment by gziolo** [comment]
Wow, this has finally landed. Thanks @stephanethomas for making sure it happened 👍

**Comment by klimeryk** [comment]
Seems it's causing some issues :) https://github.com/Automattic/wp-calypso/issues/18639#issuecomment-334925891 (or rather, it uncovered an existing bug?)

**Pull request merged**